src/zjs_util.c

```diff
@@ -309,7 +309,20 @@ void zjs_default_convert_pin(uint32_t orig, int *dev, int *pin) {
 // when accuracy isn't as important as space
 uint16_t zjs_compress_32_to_16(uint32_t num)
 {
+    if (num == 0) {
+        // GCC states that can't use __builtin_clz
+        // if x is 0, because the result is undefined
+        return 0;
+    }
+
+#ifdef __MAC__
+    // on Mac OSX, uint32_t is unsigned int
+    int zeroes = __builtin_clz(num);
+#else
+    // on Linux, uint32_t is unsigned long
```

Comment:
I think this is only in the broken SDK. So here's where just casting to (unsigned int) probably suffices? (And I guess switch to the _clz instead of _clzl.

I just wrote a sample app and showed that uint32_t is unsigned int on normal Linux, command-line default gcc.